### ISSUE TITLE
Sign-In Button not functional

### ISSUE THREAD

**Issue by ayushdev** [firebase/friendlychat-web#134]
In step 7, I edited the code as instructed in the tutorial. But after opening the app, the sign-in popup doesn't appear. Also, after opening the app, the sign-in button doesn't appear.

**Comment by Gr8Gatsby** [comment]
Button doesn't render in IE or Microsoft Edge, but it does in Chrome.

**Comment by nicolasgarnier** [comment]
Do you guys get any errors in the JavaScript console? Could you please paste them here?

Alternatively can you check that the "final" version of the app works for you? You can find it in [/web](https://github.com/firebase/friendlychat/tree/master/web)

**Comment by kolmstead** [comment]
As mentioned in #138 the index.html in web-start and web files needs the word "hidden" removed from id's user-pic, user-name, sign-out and sign-in and "disabled" needs to be removed from id submit.

Regarding #140 "Sign in doesn't do anything" I had this console error: This domain (*my domain--which happens to be a Cloud9 web IDE*) is not authorized to run this operation. Add it to the OAuth redirect domains list in the Firebase console -> Auth section -> Sign in method tab. 

Adding my domain to the firebase console as mentioned allowed the sign-in to work.

**Comment by nicolasgarnier** [comment]
@kolmstead the "hidden" attribute gets added or removed automatically depending on the user state (if a user is signed in or not). The relevant code is here: https://github.com/firebase/friendlychat/blob/master/web/scripts/main.js#L181-L195 Can you check that this code is here and running?

**Comment by kolmstead** [comment]
I've re-added the attributes and the code does work. Perhaps the initial failure (all sign-in related code being hidden) was related to the unauthorized domain redirect error.  When I have a chance I'll see if I can re-create the scenario.

**Comment by nicolasgarnier** [comment]
Cool. Yes thanks let me know if any of the steps in the codelabs are unclear. I'm always looking for making things as clear as I can.

**Comment by cassaregh** [comment]
Hi @nicolasgarnier . I followed the tutorial in the codelab about friendlychat in web but the problem is the "Sign in With google" doesn't appear in the nav bar I am stuck in step 7 in the last part where the page need to refresh.  Still the "Sign in With google" doesn't appear. I tried searching it on google and I landed on a youtube tutorial link: https://www.youtube.com/watch?v=0tM0dvllWGc&index=4&list=PLY5Ha_Gy87zjSO3w7b4CsXfaNhfTG2J8I   I followed his tutorial which is basically the same in codelabs but still it doesn't appear!!! I am literally going insane why it doesn't work on me and on the youtube tut works. I already enabled the authentication in console through email. Please help me.

**Comment by cassaregh** [comment]
Update:

I tried to deploy it skipping the step 8-12. And I went to chat-sample-a1a12.firebaseapp.com and it works! The "Sign in With google" now appears in the nav bar! What could be the problem of this? Is it the localhost? Localhost was authorized in default in console, what could it be?

**Comment by nicolasgarnier** [comment]
@cassaregh Mhh that's very strange! Can you tell me which browser (and which version) you used? I'll try to reproduce.  Also were you serving your files locally using `firebase serve` ?

**Comment by nicolasgarnier** [comment]
Can you also go back to localhost and have a look at the Javascript console to see if there are any errors?

**Comment by jbarth04** [comment]
I'm having a similar issue on step 7 of FriendlyChat with serving on localhost. I've added the specified code to index.html and main.js.  When I deploy the app using "firebase deploy", the sign-in with google button appears (see app here: https://friendlychat-bc5da.firebaseapp.com/). However, when I try to serve the page locally (on http://localhost:5000), the google sign-in button does not appear. I am using Chrome (Version 56.0.2924.87 (64-bit)) and I do not see any errors in the Javascript console. I have turned off my ad-blocker. When I run "npm run build" (as suggested on another post), I do get an npm error that package.json does not exist. Any suggestions why the sign in button does not work locally? Thanks in advance!

**Comment by nicolasgarnier** [comment]
@jbarth04 thanks for the report. This is indeed very strange behavior.

One thing that comes to mind: There could be some things leftover from other tutorials or other test apps that you've tried on localhost:5000. Could you try running the app an another port? For instance try running: `firebase serve --port 5005` and then try the app on http://localhost:5005

If things work on localhost:5005 - I'd be curious to know by the way - then the issue is probably with cookies or service workers that exist on localhost:5000, you could try to delete cookies and service workers there using the Chrome developer tools if you want to be able to test everything on the default 5000 port...

**Comment by jbarth04** [comment]
@nicolasgarnier Thanks for the suggestion, running "firebase serve --port 5005" works!  I'll have to play around with cookies for localhost:5000 (I have done a lot of demo web-apps on there).  Thanks again!

**Comment by GR0B** [comment]
I have the same issue of the signin/out buttons not showing. 
running "firebase serve --port 5005" does not work for me. 
I do not get any errors in chrome's JavaScript console. 

If I comment out the "hidden" in the index.html then get the buttons but they do not do anything when clicked.

**Comment by nicolasgarnier** [comment]
@GR0B can you deploy your code online? Run `firebase deploy` and provide the URL. I'll have a look.

**Comment by mjberman024** [comment]
Hey @nicolasgarnier - 
When i click on the Sign in with google button, an about-blank window pops up for about 1 second then immediately disappears. 
I am getting the following errors in my javascript console.
![screen shot 2017-02-23 at 4 51 12 pm](https://cloud.githubusercontent.com/assets/19418340/23280782/8de3cf76-f9e8-11e6-8057-767225f07b97.png)

**Comment by mjberman024** [comment]
When i click the sign in button the window pops up but immediately disappears. I have narrowed it down to a problem with adding my API key.  

![screen shot 2017-02-23 at 5 17 37 pm](https://cloud.githubusercontent.com/assets/19418340/23281670/426cad66-f9ec-11e6-9980-c63deef0871b.png)

**Comment by nicolasgarnier** [comment]
@bojeil-google does this last issue ring a bell? This looks like a (hopefully) temporary issue on our end.

**Comment by mjberman024** [comment]
Thank you for the quick response.  I resolved this issue on my end by creating another project and using another API Key. Here's the full console log from my previous issue. 
![screen shot 2017-02-23 at 5 30 37 pm](https://cloud.githubusercontent.com/assets/19418340/23282110/d96d3d88-f9ed-11e6-9a44-df2b9e3a913d.png)

**Comment by nadam** [comment]
Changing port to 5005 worked for me too. Can you please add that as a note in [the instructions](https://codelabs.developers.google.com/codelabs/firebase-web/#5) and/or link to this issue?

**Comment by MroSCho** [comment]
Same issue. Popup window disapears when I click the Sign-in button.
Tried changing the port to 5005, still no luck.

![console_snip](https://cloud.githubusercontent.com/assets/26675798/24324269/bca4a940-1183-11e7-9a79-22fc6dfbf44a.PNG)

Any help?

**Comment by bojeil-google** [comment]
Sorry, I didn't see this one. For sign in with popup, an iframe gets embedded. It looks like it's timing out. This may be local only. Can you test on a different network? or different device. This will help shed some light on this. Can you also check the network log to see if there is more info on this?

**Comment by MroSCho** [comment]
Thanks for quick reply.
Tried on 2 other networks, same problem.
So I went on the finish all the steps and deploy. Using the deploy link, all worked well on my iPhone and Nexus 6p, but not on my Surface.... So the logical next step was to check if Bitdefender was being too strict. Turned off the web protection and everything went back to normal, even localhost works again.... stupid me.
Thanks again!

**Comment by bojeil-google** [comment]
Thanks MroSCho for investigating this and getting back to us on this.

**Comment by secorbett13** [comment]
I am having the same problem when using Edge or IE 11. If I remove the "position: absolute" attribute on the user-container CSS definition, the area shows up and all is well.

**Comment by csaron92** [comment]
I had the same issue as some of the guys previously (sign-in button appears but the popup immediatly disappers) and the solution was to enable the authentication in Firebase Console.

**Comment by raviprasadmr** [comment]
After Step 10, I am totally stuck. No Sign-in button.

**Comment by KairusDarkSeeker** [comment]
@ayushdev The only solution for this is to not enable URL the single-app page option during init

Just like the options I chose shown on this photo:
![Options](https://i.imgur.com/CYNDHwS.jpg)

**Comment by jgransky** [comment]
I had this issue too but just resolved it. The two things I did were (1) enable the Google Identity Toolkit (https://console.developers.google.com/apis/api/identitytoolkit.googleapis.com/overview?project=xxxxx where xxxxx is the specific project number), and (2) run `firebase init` in the project folder. Not sure which of those two was responsible for fixing it, but now it works.

**Comment by neerajraghuwanshi** [comment]
same problem i have (In step 7, I edited the code as instructed in the tutorial. But after opening the app, the sign-in popup doesn't appear. Also, after opening the app, the sign-in button doesn't appear.)
errors;::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
firebase-auth.js:332 Uncaught Error: Cannot instantiate firebase-auth.js - be sure to load firebase-app.js first.
    at firebase-auth.js:332
23:51:44.376 main.js:59 Uncaught TypeError: firebase.auth is not a function
    at FriendlyChat.initFirebase (main.js:59)
    at new FriendlyChat (main.js:53)
    at window.onload (main.js:261)
23:52:38.047 main.js:166 Uncaught TypeError: Cannot read property 'currentUser' of undefined
    at FriendlyChat.checkSignedInWithMessage (main.js:166)
    at FriendlyChat.saveMessage (main.js:75)

**Comment by nicolasgarnier** [comment]
I've reworked the codelab to make it more simple and obvious what should be copy-pasted/added to the code. Hopefully this fixes most issues like these ones.

**Comment by tanseerpasha** [comment]
I was able to fix the issue by adding the  initialization snippet (which you can find in the Firebase console of your project, in the Project Overview > Add Firebase to your Web App: ) into the <head>tag (I added just before </head> tag)

(Found the clue here https://firebase.google.com/docs/web/setup 
3.Copy and paste your project's customized code snippet in the <head> tag of your page, before other script tags.)

**Comment by nicolasgarnier** [comment]
Normally you should not need to paste your initialisation snippets. It is automatically included at that line:

https://github.com/firebase/friendlychat-web/blob/master/web-start/index.html#L118

`/__/firebase/init.js` should contain your initialisation snippet if you are serving your app with the Firebase CLI (e.g. with `firebase serve`) and you've correctly setup the app before using `firebase use --add`

You do need to manually add the initialisation snippets from **Project Overview > Add Firebase to your Web App** only if you are serving your app using your own web-server (other than firebase servers or deployed firebase hosting)

**Comment by dbkellogg** [comment]
I've been working on this friendly chat project off & on for a few weeks now. I tried it on an old Windows 7 PC and got stuck on Step 7, as per this discussion (Google login popup would not show up). Deciding that it was probably an issue with my PC, I upgraded to a Win10 system and started over from scratch. Upon reaching Step 7, I've had the same experience again. I'm using the latest chrome browser, deploying from VS Code terminal, and the only terminal output I get is:
"Warning: You're using Node.js v10.10.0 but Google Cloud Functions only supports v6.11.5.
i  functions: No HTTPS functions found. Use firebase functions:shell if you would like to emulate other types of functions."

I tried switching ports (though this is a new PC so I doubt there's any issue with cookies or caching), verifying my API key, and fully deploying the app. I went back through all the steps to verify I'd done everything exactly as recommended. I cannot get the sign-in dialog to show. What should I try next?

**Comment by nicolasgarnier** [comment]
@dbkellogg Have you checked the JavaScript console to see if there are any errors?

**Comment by tanseerpasha** [comment]
@nicolasgarnier 
Thanks. But with default snippet, sign in was not working. It wasn't showing any accounts. What i did was, moved the line /__/firebase/init.js with in the head tag. As this fixed the issue, I completed the rest of the code lab(but notifications didn't work). I actually wanted to implement  notifications for android which I was able to. Thanks the code lab was definitely very useful.

**Comment by dbkellogg** [comment]
Is that the same as the debug console? I'm a bit new to all of this.
In the debug console I have a `ReferenceError: window is not defined` for:
`function checkSetup() {
  if (!window.firebase || !(firebase.app instanceof Function) || !firebase.app().options) {
    window.alert('You have not configured and imported the Firebase SDK. ' +
        'Make sure you go through the codelab setup instructions and make ' +
        'sure you are running the codelab using `firebase serve`');
  }
}`

**Comment by egdiala** [comment]
In step 7, the sign in button doesn't respond when I click on it. Please I need help, I'm new to this.

**Comment by Isonomic** [comment]
When I click Sign in it opens briefly and then closes. This is late 2019 lol.  What the hell Google. Here is my error
![Screenshot (163)](https://user-images.githubusercontent.com/37713396/71328662-3e996e80-24d8-11ea-969d-ab171798a4bb.png)

**Comment by jhuleatt** [comment]
@Isonomic it looks like things are working as intended (the `auth/web-storage-unsupported` error code is documented [here](https://firebase.google.com/docs/reference/js/firebase.auth.Error)).

The Firebase Auth SDK requires 3p cookies and localstorage, so if you've disabled those, it won't work. If you cannot enable them, a potential workaround is discussed [here](https://github.com/firebase/firebase-js-sdk/issues/865#issuecomment-392117643).

If you have any more questions, please open a new issue.

**Comment by austinedger0811** [comment]
If you go to your Authorization tab in firebase, does it show users added?

**Comment by hyc0812** [comment]
@nicolasgarnier 
Thank you for the nice tutorial!

My issue is that: 
I actually can log in with my google account (although the login window pops up and disappears), and I can send text messages and pictures, and I also can find messages and pictures in my firestore database and storage, which seems fine.

But the upright corner never changes even after my login:
<img width="176" alt="Screen Shot 2022-01-27 at 10 23 41 AM" src="https://user-images.githubusercontent.com/80454212/151400574-614abee2-77b1-488b-a33e-02dfe9b81306.png">

The chat web page never displays my words and pictures that have been sent. By which I mean, these pieces of stuff have been sent to my firestore database and storage but never display on the webpage:

https://user-images.githubusercontent.com/80454212/151402036-c13af525-f879-49ac-8e95-7456ec060c3f.mov

Check it in firestore database:
![Screen Shot 2022-01-27 at 10 41 44 AM](https://user-images.githubusercontent.com/80454212/151403894-0e54c15f-6d63-4924-bb51-f145894a8d8b.png)


The inspection:
![Screen Shot 2022-01-27 at 10 37 59 AM](https://user-images.githubusercontent.com/80454212/151403358-7045548a-e09c-4f8d-be26-4b37272455a5.png)

And lastly, this is the link to my demo:
https://yong-project-001.web.app/


Thank you for your help!

**Comment by nicolasgarnier** [comment]
@hyc0812 I think in your code: Line 12 > `const firebaseConfig = {` should be `const config = {`

**Comment by hyc0812** [comment]
@nicolasgarnier Thank you for your quick response! 

I have fixed this error by modifying `config` to `firebaseConfig` in the function `getFirebaseConfig()`, and everything works well. Thank you!

![Screen Shot 2022-01-27 at 4 07 46 PM](https://user-images.githubusercontent.com/80454212/151451793-cec4be03-9579-4ae2-85c2-77d3e524adfa.png)

**Comment by damianohienmhen** [comment]
I actually have a good solution for this problem. If you hit the inspect button on your Chrome browser, you'll see in the console that you need to manually add the domain for the pop up in the firebase console. The console logs will clearly tell you what domain to add and where to add it in your firebase console. Fixed my sign in problem easily.

**Comment by puskarwagle** [comment]
Uncaught ReferenceError: config is not defined at webpack-internal:///./src/firebase-config.js:23:3

I get the above error in my console. Clicking on the "SIGN-IN WITH GOOGLE" button does nothing. The button was always visible and not hidden for me. I modified config to firebaseconfig. It was not helpful.

**Comment by AnjaliSuman3011** [comment]
@nicolasgarnier  hey can you please help me out  with the firebase project app..im stuck at step 7 ..the RSVP button does not gets clicked and nothing further happnens
